frontend/src/App.tsx

```diff
@@ -64,32 +64,43 @@ const App = (): React.ReactElement => {
                 <Route path={Routes.HOME_PAGE} element={<Default />} />
                 <Route
                   path={Routes.CREATE_ENTITY_PAGE}
-                  element={<CreatePage />}
-                />
+                  element={<PrivateRoute />}
+                >
+                  <Route path="" element={<CreatePage />} />
+                </Route>
```

Comment:
Although the way you guys set this up looks great, I have a proposal that makes things just a bit cleaner IMO.

Set up `PrivateRoute.tsx` as so:

```
import React, { useContext } from "react";
import { Navigate } from "react-router-dom";

import AuthContext from "../../contexts/AuthContext";
import { LOGIN_PAGE } from "../../constants/Routes";

const PrivateRoute: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => {
  const { authenticatedUser } = useContext(AuthContext);

  return authenticatedUser ? children : <Navigate to={LOGIN_PAGE} />;
};

export default PrivateRoute;
```

and within `App.tsx`, set up the private routes like this:

```
<Route
                  path={Routes.CREATE_ENTITY_PAGE}
                  element={
                    <PrivateRoute>
                      <CreatePage />
                    </PrivateRoute>
                  }
                />
```

Thoughts on this?

Reply:
This looks great, I wasn't really sure how to make this more clean so I left it as is, but I think this is a much better approach. I'll update this